File: stasis-core/src/de/boxxit/stasis/AuthenticationMissmatchException.java
package de.boxxit.stasis;

import com.esotericsoftware.kryo.DefaultSerializer;
import de.boxxit.stasis.serializer.AuthenticationMissmatchExceptionSerializer;

/**
 * User: Christian Fruth
 */
@DefaultSerializer(AuthenticationMissmatchExceptionSerializer.class)
public class AuthenticationMissmatchException extends Exception
{
	private boolean authenticated;

	public AuthenticationMissmatchException(boolean authenticated)
	{
		this.authenticated = authenticated;
	}

	public boolean isAuthenticated()
	{
		return authenticated;
	}
}

File: stasis-ios/Stasis/AuthenticationMissmatchException.m
//
//  AuthenticationMissmatchException.m
//  Stasis
//
//  Created by Christian Fruth on 06.03.13.
//  Copyright (c) 2013 Boxx IT Solutions e.K. All rights reserved.
//

#import "AuthenticationMissmatchException.h"
#import "AuthenticationMissmatchExceptionSerializer.h"

@implementation AuthenticationMissmatchException

+ (instancetype)exceptionWithAuthenticated:(BOOL) authenticated
{
	return [[AuthenticationMissmatchException alloc] initWithAuthenticated:authenticated];
}

- (id)initWithAuthenticated:(BOOL) authenticated
{
	self = [super initWithName:@"StasisException" reason:@"Authentication missmatched" userInfo:nil];
	
	if (self != nil)
	{
		_authenticated = authenticated;
	}
	
	return self;
}

+ (NSString *)serializingAlias
{
	return @"de.boxxit.statis.AuthenticationMissmatchException";
}

+ (Class)defaultSerializer
{
	return [AuthenticationMissmatchExceptionSerializer class];
}

@end
